airflow/www/views.py

```diff
@@ -2685,6 +2685,9 @@ class XComView(wwwutils.SuperUserMixin, AirflowModelView):
     filter_converter = wwwutils.UtcFilterConverter()
     form_overrides = dict(execution_date=DateTimeField)
 
+    def on_model_change(self, form, model, is_created):
+        XCom.set(**form.data)
```

Comment:
@ashb @XD-DENG I am wondering if passing `**form.data` is a security vulnerability or not.

Would it then be possible to add a html field to the form called `session` to hijack a session? Should I rather set it directly `task_id=form.data['task_id']` etc. or even use model's values `task_id=model.task_id` etc. ? What is the difference? 

It works but there are many points I am unsure about.

Reply:
I think that this includes the fields are known to the form, so this should be okay.

But this will end up storing two records - one that we create here, and another from https://github.com/flask-admin/flask-admin/blob/8341a9ceae6e9206cb6b91a3b896ed02faaafccc/flask_admin/contrib/sqla/view.py#L1129-L1132

Reply:
The "correct" fix is probably to make the `value` accessors on XCom transparently handle the (de)serialization.

Reply:
I see. 

At first I tried to set `model.value = model.value.encode('utf-8')` described [here](https://github.com/flask-admin/flask-admin/issues/1437#issuecomment-392118807) but that raised `_pickle.UnpicklingError: invalid load key, 'v'.` :/

Reply:
> But this will end up storing two records - one that we create here, and another from https://github.com/flask-admin/flask-admin/blob/8341a9ceae6e9206cb6b91a3b896ed02faaafccc/flask_admin/contrib/sqla/view.py#L1129-L1132

No, there will actually only be one entry in the table because of `# remove any duplicate XComs` section in `XCom.set`.

> The "correct" fix is probably to make the `value` accessors on XCom transparently handle the (de)serialization.

I don't know how this can be done. So for now I just copied the block for the string to byte conversion to the `on_model_change`. It might not be the best solution but this at least does not 
* creates a session
* removes any duplicate XComs
* insert new XCom

(what `XCom.set(**form.data)` did)

@ashb PTAL

Reply:
Ah okay so this is good enough for now then I guess.